prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -479,7 +498,9 @@ class FHIRRouter(
             }
         }
 
-        return if (exceptionFilters.isNotEmpty()) {
+        return if (singlePass && successFilters.isNotEmpty()) {
+            Pair(true, null)
+        } else if (exceptionFilters.isNotEmpty()) {
             Pair(false, "(exception found) $exceptionFilters")
         } else if (reverseFilter) {
```

Comment:
How does `reverseFilter` work with `singlePass`? For condition filters it's not possible for them both to be true currently, but hypothetically if that were to happen, the logic here gets kind of confusing: `reverseFilter` reverses the result, but not if we find an exception, and not if `singlePass` is true and we find a passing condition.

Reply:
I was also thinking of that one. I would think that the reverse of singlePass is `if any fail, it's false`

Reply:
I'm gonna go with the latter of the above, and then we can go into more detail while discussing the overall complexity of the method